engines/nancy/action/actionmanager.cpp

```diff
@@ -0,0 +1,466 @@
+                        if (record->days == -1) {
+                            record->days = NancySceneState._timers.playerTime.getDays();
+                            dep.satisfied = true;
+                            break;
+                        }
+
+                        if (record->days < NancySceneState._timers.playerTime.getDays()) {
+                            record->days = NancySceneState._timers.playerTime.getDays();
+                            for (uint j = 0; j < record->dependencies.size(); ++j) {
+                                if (record->dependencies[j].type == kPlayerTime) {
+                                    record->dependencies[j].satisfied = false;
+                                }
+                            }
+                        }
+
+                        break;
+                    case kUseItem: {
+                        bool hasUnsatisfiedDeps = false;
+                        for (uint j = 0; j < record->dependencies.size(); ++j) {
+                            if (j != i && record->dependencies[j].satisfied == false) {
+                                hasUnsatisfiedDeps = true;
+                            }
+                        }
+
+                        if (hasUnsatisfiedDeps) {
+                            break;
+                        }
+
+                        record->itemRequired = dep.label;
+
+                        if (dep.condition == 1) {
+                            record->itemRequired += 100;
+                        }
+                        
+                        dep.satisfied = true;
+                        break;
+                    }
+                    case kTimeOfDay:
+                        if (dep.label == (byte)NancySceneState._timers.timeOfDay) {
+                            dep.satisfied = true;
+                        }
+
+                        break;
+                    case kTimerNotDone:
+                        if (NancySceneState._timers.timerTime <= dep.timeData) {
+                            dep.satisfied = true;
+                        }
+
+                        break;
+                    case kTimerDone:
+                        if (NancySceneState._timers.timerTime > dep.timeData) {
+                            dep.satisfied = true;
+                        }
+
+                        break;
+                    case kDifficultyLevel:
+                        if (dep.condition == NancySceneState._difficulty) {
+                            dep.satisfied = true;
+                        }
+
+                        break;
+                    default:
+                        break;
+                    }
+                }
+            }
+
+            // An orFlag marks that its corresponding dependency and the one after it
+            // mutually satisfy each other; if one is satisfied, so is the other
+            for (uint i = 1; i < record->dependencies.size(); ++i) {
+                if (record->dependencies[i-1].orFlag) {
```

Comment:
Formatting is wrong here should be `i - 1`

Reply:
Fixed.